### PR TITLE
#69 add default hours for event

### PR DESCRIPTION
**Pull request by oookli**

### CODE REVIEW

File: app/controllers/admin/events_controller.rb [hunk 1]
```diff
@@ -3,7 +3,10 @@ class EventsController < BaseController
     helper_method :event, :events
 
     def new
-      @event = Event.new
+      @event = Event.new(
+        started_at: date_with_specified_hour(Event::DEFAULT_STARTED_AT_HOURS),
```
Comment (sveredyuk) [inline app/controllers/admin/events_controller.rb]: this looks like business logic in controller. Bad.
Try extract this logic to service like `@event = Event::New.call` and let service to rule logic

Comment (oookli) [reply app/controllers/admin/events_controller.rb]: ok, thanks, I'll fix it

Comment (oookli) [reply app/controllers/admin/events_controller.rb]: @sveredyuk I've updated it, please check if everything is ok

File: app/services/event/new.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      Event.new(
+        started_at: date_with_specified_hour(Event::DEFAULT_STARTED_AT_HOURS),
+        finished_at: date_with_specified_hour(Event::DEFAULT_FINISHED_AT_HOURS)
+      )
+    end
+
+    private
+
+    def date_with_specified_hour(hour)
+      DateTime.now.change({hour: hour, min: 0, sec: 0})
```
Comment (kushniryb) [inline app/services/event/new.rb]: 1. There's no need to use `{}` here
2. Check the method [source](http://apidock.com/rails/DateTime/change) - you can just write `DateTime.now.change(hour: 0)`, minutes and seconds will be 0 by default

File: app/services/event/new.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Event
+  class New
+
```
Comment (kushniryb) [inline app/services/event/new.rb]: Remove space here

Comment (kushniryb) [inline app/services/event/new.rb]: There's no need for an empty line here

File: app/services/event/new.rb (hunk 2 above)
Comment (kushniryb) [reply app/services/event/new.rb]: Also, use `Time.zone.now`. Article for [reference](http://danilenko.org/2012/7/6/rails_timezones/)

File: app/services/event/new.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    end
+
+    private
+
+    def date_with_specified_hour(hour)
+      DateTime.now.change({hour: hour, min: 0, sec: 0})
+    end
+
+  end
+end
```
Comment (kushniryb) [inline app/services/event/new.rb:17]: Also, I'm not sure it's worth creating a separate class which just sets 2 attributes, maybe an `after_initialize`hook would be better? What do you think @sveredyuk @denysmedynskyi

**Review by kushniryb — CHANGES_REQUESTED** [review]
Please, fix a few issues described above

File: app/services/event/new.rb (hunk 3 above)
Comment (oookli) [reply app/services/event/new.rb]: okok

File: app/services/event/new.rb (hunk 2 above)
Comment (oookli) [reply app/services/event/new.rb]: thanks @kushniryb , I'll fix this issues

File: app/services/event/new.rb (hunk 4 above)
Comment (oookli) [reply app/services/event/new.rb:17]: @kushniryb but `after_initialize` not very transparently, I think

Comment (denys-medynskyi) [reply app/services/event/new.rb:17]: Callbacks are not good for health. I prefer not to use `after_initialize`

File: app/services/event/new.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+class Event
+  class New
```
Comment (sveredyuk) [inline app/services/event/new.rb]: inherit form ApplicationService

File: app/services/event/new.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    def self.call
+      new.call
```
Comment (sveredyuk) [inline app/services/event/new.rb]: ApplicationService class inheritance gives you this factory method for free

File: app/services/event/new.rb (hunk 4 above)
Comment (sveredyuk) [reply app/services/event/new.rb:17]: agree with @denysmedynskyi callbacks are evil. Clear invocation are better than magical `after_some_wtf_when_where_why_shit`

Comment (kushniryb) [reply app/services/event/new.rb:17]: Yeah, callbacks are evil, I agree, but creating this kind of an object to populate 2 fields seems like an overkill to me. Also, why not Form Object here?

Comment (sveredyuk) [reply app/services/event/new.rb:17]: @kushniryb FormObject will rule this in version when we rewrite all this to Trailblazer and than cut to engines )

**Comment by oookli** [comment]
I have updated this branch already, thanks for responses

**Review by sveredyuk — APPROVED** [review]
Resolve conflicts
Improve test cove up to 100%

**Comment by oookli** [comment]
@sveredyuk Thanks
image
we've updated this values, we should use some datetimepicker in future for this?

**Comment by sveredyuk** [comment]
@oookli Ofc #158

**Comment by oookli** [comment]
@sveredyuk Could I take that?

**Review by kushniryb — APPROVED** [review]

**Comment by sveredyuk** [comment]
@oookli does it ready for merge ?

**Comment by oookli** [comment]
@sveredyuk yes, I think

**Comment by sveredyuk** [comment]
@oookli merge

**Comment by oookli** [comment]
@sveredyuk thanks

**Pull request merged**